src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -480,27 +636,141 @@ export const Deposit: React.FC<DepositProps> = ({
       if (isBelowOutboundFee)
         return translate('trade.errors.amountTooSmall', { minLimit: outboundFeeLimit })
 
-      const cryptoBalancePrecision = bnOrZero(balance).div(bn(10).pow(asset.precision))
-      const valueCryptoPrecision = bnOrZero(value)
-      const hasValidBalance =
-        cryptoBalancePrecision.gt(0) &&
-        valueCryptoPrecision.gt(0) &&
-        cryptoBalancePrecision.gte(value)
-      if (valueCryptoPrecision.isEqualTo(0)) return ''
-      return hasValidBalance || 'common.insufficientFunds'
+      // Balance / Fees / Sweep fees check outside of react
+      // These are very boilerplaty ~100 LoC checks, but ensure we use the react-query queries and the react-query cache
+      // We cannot use the react-query hooks here because of a catch-22 situation: the hooks are reactive,
+      // but by the time the validation methods are hit, hooks didn't react on input values just yet and only will a few render cycles later
```

Comment:
We may be able to keep this whole escape-hatched logic DRY.
The actual hooks are reactive, have `enabled` options and  dependencies on each other so are preferred at view-layer for actual reactivity, but in the case of the crypto/fiat validation methods, the logic should be almost a 1/1 copypasta.

Reply:
https://github.com/shapeshift/web/pull/5614/commits/b5cf3b39fd2608aa49a860dee8f36987d1082778